tests/test_chain_db.py

```diff
@@ -0,0 +1,141 @@
+from regolith.chained_db import ChainDB
+
+
+def test_dddi():
+    a = {'a': {'a': {'a': 1}}}
+    z = ChainDB(a)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['a'], ChainDB)
+    assert isinstance(z['a']['a']['a'], int)
+    assert z['a']['a']['a'] + 1 == 2
+
+
+def test_second_mapping():
+    m1 = {'a': {'m': {'x': 0}}}
+    m2 = {'a': {'m': {'y': 1}}}
+    z = ChainDB(m1)
+    z.maps.append(m2)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'].maps, list)
+    assert z['a']['m']['y'] == 1
+
+
+def test_double_mapping():
+    m1 = {'a': {'m': {'y': 0}}}
+    m2 = {'a': {'m': {'y': 1}}}
+    z = ChainDB(m1)
+    z.maps.append(m2)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'].maps, list)
+    assert isinstance(z['a']['m']['y'], int)
+    assert z['a']['m']['y'] == 1
+
+
+def test_list_mapping():
+    m1 = {'a': {'m': 'x'}}
+    m2 = {'a': {'m': 'y'}}
+    z = ChainDB(m1)
+    z.maps.append(m2)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'], str)
+    assert z['a']['m'] == 'y'
+
+
+def test_mixed_mapping():
+    m1 = {'a': {'m': {'y': 1}}}
+    m2 = {'a': {'m': 1}}
+    z = ChainDB(m1)
+    z.maps.append(m2)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'], int)
+    assert z['a']['m'] == 1
+
+
+def test_exactness():
+    d = {'y': 1}
+    m1 = {'a': {'m': d}}
+    z = ChainDB(m1)
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'], ChainDB)
+    assert isinstance(z['a']['m'].maps[0], dict)
+    assert d is z['a']['m'].maps[0]
+
+
+def test_exactness_setting():
+    d = {'y': 1}
+    m1 = {'a': {'m': d}}
+    z = ChainDB(m1)
+    e = {'z': 2}
+    z['a']['m'] = e
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'], ChainDB)
+    assert isinstance(z['a']['m'].maps[0], dict)
+    assert e is z['a']['m'].maps[0]
+
+
+def test_exactness_setting_multi():
+    d = 'a'
+    e = 'b'
+    m1 = {'a': {'m': d}}
+    m2 = {'a': {'m': e}}
+    z = ChainDB(m1)
+    z.maps.append(m2)
+    g = 'c'
+    z['a']['m'] = g
+    assert isinstance(z['a'], ChainDB)
+    assert isinstance(z['a']['m'], str)
+    assert isinstance(z['a'].maps[0], dict)
+    assert g is z['a'].maps[1]['m']
+    # We sent this to the first map not the last
+    assert z['a']['m'] is g
```

Comment:
This is not a great comparison for a sort of subtle reason.  Namely, CPython will intern (cache) string. So two string objects (normally literals) that are independently created may, in fact, be the same object. Here for example, 

```python
>>> 'c' is 'c'
True
```

It is better to do this kind of test with some kind of container type.  Even tuples work, because

```python
>>> ('c',) is ('c',)
False
```